pkilint/etsi/etsi_shared.py

```diff
@@ -147,3 +149,111 @@ def validate(self, node):
 VALIDATION_INTERNAL_IP_ADDRESS = validation.ValidationFinding(
     validation.ValidationFindingSeverity.ERROR, "etsi.internal_ip_address"
 )
+
+
+class LegalPersonAttributeAllowanceValidator(
+    common.AttributeIdentifierAllowanceValidator
+):
+
+    def __init__(self, code_classifier, required_attributes, path_to_rdnSequence):
+        _ATTRIBUTE_ALLOWANCES = {a: Rfc2119Word.MUST for a in required_attributes}
+        super().__init__(
+            _ATTRIBUTE_ALLOWANCES,
+            code_classifier,
+            Rfc2119Word.MAY,
+            path=path_to_rdnSequence,
+        )
+
+
+class LegalPersonDuplicateAttributeAllowanceValidator(validation.Validator):
+    """
```

Comment:
These citation comments should probably be moved to the appropriate sub-class so they document the validation finding declaration.